packages/main/src/SelectPopover.hbs

```diff
@@ -38,9 +38,23 @@
 			{{/if}}
 		{{/unless}}
 
-		<ui5-list mode="SingleSelectAuto" separators="None" @keydown="{{_handlePickerKeydown}}" @ui5-selection-change="{{_handleSelectionChange}}" @ui5-item-press="{{_handleItemPress}}">
+		<ui5-list
+			id="{{_id}}-select-list"
+			mode="SingleSelectAuto"
+			separators="None"
+			@keydown="{{_handlePickerKeydown}}"
+			@ui5-selection-change="{{_handleSelectionChange}}"
+			@ui5-item-press="{{_handleItemPress}}"
+		>
 			{{#each _syncedOptions}}
-				<ui5-li ?selected="{{this.selected}}" icon="{{this.icon}}" id="{{this.id}}-li">
+				<ui5-li
+					id="{{this.id}}-li"
+					icon="{{this.icon}}"
+					?selected="{{this.selected}}"
+					?aria-selected="{{this.selected}}"
```

Comment:
The list item sets aria-selected internally (to the root tag), depending on its selected property - not sure what happens if we set it on the ui5-li tag too, maybe nothin and I think we dont have to add it at ll. For the aria-setsize and aria-posinset attrs,  I remember that at least VoiceOver (not sure about JAWS) announce the position and the total size without setting, but if we have to set them, should we set them on root tag? Because in the issue, the author points to the ShadowDom "li" element. It will be best to consult with our local ACC expert.

Reply:
all aria attributes should be set on the element who has the focus (in your case the list item and the div with class ui5-select-root) For the list items and the aria-setsize and aria-posinset - they should be read out by the screen reader since they have a role=option.

I read the issue which is reported, but I don't see a point in implementing most of the attributes. What is more the ui5-select is implemented following the aria 1.1 best practice examples for a collapsible listbox:
https://www.w3.org/TR/wai-aria-practices-1.1/examples/listbox/listbox-collapsible.html

IMO:
aria-roledescription - could be added a roledescription of "Select" like in UI5
aria-disabled - already implemented
aria-readonly - no need, you don't have a read only property
aria-expanded - could be added
aria-controls - could be added, but I don't see the need we have a aria-haspopup=listbox (what is more you have to point to the element with role listbox and I am not sure if this is possible)
aria-activedescendant - could not be implemented since the items are in a different DOM, the only thing that could be done is a custom text like in the ui5-input (List item x of y is selected <li-name>)
aria-selected - should be set to the li
aria-setsize/aria-posinset- check if it is spoken, if not - the only solution is the one mentioned in aria-activedescendant

Hope that helps.

Reply:
Done